src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -835,28 +836,83 @@ export default class Fms {
      * @param routeString {string}
      * @return {boolean}
      */
-    replaceDepartureProcedure(routeString) {
-        const routeStringElements = routeString.split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
+    replaceDepartureProcedure(routeString, airportIcao) {
+        const routeStringElements = routeString.toUpperCase().split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
 
-        if (routeStringElements.length !== 3) {
+        if (routeStringElements.length > 3) {
             return [false, 'departure procedure format not understood'];
         }
 
-        const procedureId = routeStringElements[1];
+        let procedureId = routeStringElements[0];
 
-        if (!NavigationLibrary.hasProcedure(procedureId)) {
+        if (routeStringElements.length === 3) {
+            procedureId = routeStringElements[1];
+        }
+
+        const sidModel = NavigationLibrary.getProcedure(procedureId);
+
+        if (_isNil(sidModel)) {
             return [false, `unknown procedure "${procedureId}"`];
         }
 
-        const wasSuccessful = this._routeModel.replaceDepartureProcedure(routeString);
+        if (routeStringElements.length === 1) { // RouteString looks like PROC
+            const exitPoint = _findLast(this.waypoints, (waypointModel) => sidModel.hasExit(waypointModel.name));
 
-        if (wasSuccessful) {
-            this._updateDepartureRunwayFromRoute();
+            if (!exitPoint) {
+                return [false, `the ${procedureId.toUpperCase()} departure doesn't have an exit along our route`];
+            }
 
-            return [true, ''];
+            routeStringElements.push(exitPoint.name);
         }
 
-        return [false, `route of "${routeString}" is not valid`];
+        if (routeStringElements.length === 2) { // RouteString looks like PROC.EXIT
+            const runwayModel = this.departureRunwayModel;
+
+            if (_isNil(runwayModel)) {
+                return [false, 'unsure if we can accept that procedure; we don\'t have a runway assignment'];
+            }
+
+            let entryName = sidModel.getFirstEntryPoint();
+
+			
+			
+			
+					
+			if (this.currentPhase!==FLIGHT_PHASE.APRON){
```

Comment:
yikes, this whole section needs some formatting love.  Be sure you have both editorconfig and eslint plugins installed in your editor/ide.  Also be sure your code looks like existing code.

I'll hold off an any additional reviews for this section until after it's been formatted correctly

Reply:
done